### PR TITLE
Make datasource variables multiselect and dashboard repeatable

### LINKED ISSUE #7492: Implement multi-value for datasource
### Request
Hi, first thank you for your amazing work !
After reading and testing some things I have founded a question (see the source) and I have decided to make this request. 

The request is to allow template variables `type datasource` to pull from multiple data sources. Like we can do with custom for example.

### Use case
We have multiple DB data sources and we want to view in one dashboard that uses templating (dynamic row and panel repeats) all the data. This is to create a dashboard "overview" for example.

### How
By making `datasource` drop down menu multi-value.

### source
 It's linked with the issue #3480 who is 2 years old.

### PR DESCRIPTION
**Pull request by jdoupe**
closes #7492 (and #7030)

### CODE REVIEW

File: public/app/features/panel/metrics_panel_ctrl.ts [hunk 1]
```diff
@@ -79,9 +79,28 @@ class MetricsPanelCtrl extends PanelCtrl {
...
+
+    // if a data source variable was found, use its value
+    if (datasourceVarName !== '' && this.panel.scopedVars && this.panel.scopedVars[datasourceVarName]) {
+      mydatasource = this.panel.scopedVars[datasourceVarName].value;
+    }
+
     // load datasource service
     this.datasourceSrv
-      .get(this.panel.datasource)
+      .get(mydatasource)
```
Comment (torkelo) [inline public/app/features/panel/metrics_panel_ctrl.ts]: would it not be easier to skip all the added code above and pass scopedVars to datasourceSrv.get ?

File: public/app/features/plugins/datasource_srv.ts [hunk 2]
```diff
@@ -40,6 +40,10 @@ export class DatasourceSrv {
   }
 
   loadDatasource(name: string): Promise<DataSourceApi> {
+    // if there are multiple datasources provided, just use the first one
+    const re = /{([^,}]+).*/;
+    name = name.replace(re, '$1');
```
Comment (torkelo) [inline public/app/features/plugins/datasource_srv.ts]: really hate this. Feels so hacky. The problem of only being able to support multi if you have configured repeating is one of the biggest reasons I am not sure this is a good idea. 

See discussions here: https://github.com/grafana/grafana/pull/10849

File: public/app/features/panel/metrics_panel_ctrl.ts (hunk 1 above)
Comment (jdoupe) [reply public/app/features/panel/metrics_panel_ctrl.ts]: I just decided to learn enough ts to hack this change out as we desperately need to stop littering our dashboards with duplicated, "hardcoded" datasources.  I agree that moving the code to datasource_srv would be preferable, but this isn't the only place datasourceSrv.get() is called, and typescript's method overloading options seem a bit weak.

File: public/app/features/plugins/datasource_srv.ts (hunk 2 above)
Comment (jdoupe) [reply public/app/features/plugins/datasource_srv.ts]: Agreed.  I felt a little dirty writing it, but necessity overruled my concerns, and knowing/documenting the side effect seems reasonable.  I'm not sure why one would ever enable multiselect on a datasource unless they intended on doing repeat on it.

File: public/app/features/panel/metrics_panel_ctrl.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/panel/metrics_panel_ctrl.ts]: > and typescript's method overloading options seem a bit weak. 

??? 

Well it has do be done. It cannot be done here.

**Comment by torkelo** [comment]
Refactored this PR and opened another PR, https://github.com/grafana/grafana/pull/15914

File: public/app/features/panel/metrics_panel_ctrl.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/panel/metrics_panel_ctrl.ts]: think you did not understand what I meant, you can see my refactoring & simplification here: https://github.com/grafana/grafana/pull/15914/files

**Pull request closed without merging**